src/middlewared/middlewared/plugins/system.py

```diff
@@ -96,11 +96,16 @@ class Config:
                     f'{schema}.serialport',
                     'Please specify a serial port when serial console option is checked'
                 )
-            elif serial_choice not in await self.serial_port_choices():
-                verrors.add(
-                    f'{schema}.serialport',
-                    'Serial port specified has not been identified by the system'
-                )
+            else:
+                data['serialport'] = serial_choice = hex(
+                    int(serial_choice)
+                ) if serial_choice.isdigit() else serial_choice
+
+                if serial_choice not in await self.serial_port_choices():
+                    verrors.add(
+                        f'{schema}.serialport',
+                        'Serial port specified has not been identified by the system'
+                    )
```

Comment:
What is an example of decimal input port?

Reply:
`1016`

Reply:
How is that a decimal?

Reply:
You mean integer?

Reply:
Yup, base 10

Reply:
Ok, lets reword the ticket/commit please, decimal is very confusing